core-library/src/main/java/org/silverpeas/core/index/search/SimpleSearchEngine.java

```diff
@@ -108,4 +120,100 @@ private boolean isSpellingNeeded(List<MatchingIndexEntry> results) {
     SearchCompletion completion = new SearchCompletion();
     return completion.getSuggestions(keywordFragment);
   }
+
+  private List<MatchingIndexEntry> filterMatchingIndexEntries(
+      List<MatchingIndexEntry> matchingIndexEntries, String userId) {
+    if (matchingIndexEntries == null || matchingIndexEntries.isEmpty()) {
+      return new ArrayList<>();
+    }
+    List<MatchingIndexEntry> results = new ArrayList<>(matchingIndexEntries.size());
+    ComponentAuthorization authorization = null;
+    try {
+      authorization = getSecurityIntf();
+      authorization.enableCache();
+    } catch (Exception e) {
+      SilverLogger.getLogger(this).error(e.getMessage(), e);
+    }
+
+    for (MatchingIndexEntry result : matchingIndexEntries) {
+      if (!isMatchingIndexEntryAvailable(result, userId, authorization)) {
+        continue;
+      }
+      results.add(result);
+    }
+
+    authorization.disableCache();
+
+    return results;
+  }
+
+  private boolean isMatchingIndexEntryAvailable(MatchingIndexEntry mie, String userId,
+      ComponentAuthorization authorization) {
+    // Do not filter and check external components
+    if (enableExternalSearch && isExternalComponent(mie.getServerName())) {
+      mie.setExternalResult(true);
+      // Filter only Publication and Node data
+      String objectType = mie.getObjectType();
+      if ("Versioning".equals(objectType) || "Publication".equals(objectType)
+          || "Node".equals(objectType)) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    String componentId = mie.getComponent();
+    if (componentId.startsWith("kmelia")) {
+      return authorization
+          .isObjectAvailable(componentId, userId, mie.getObjectId(), mie.getObjectType());
+    }
+    // contrôle des droits sur les espaces et les composants
+    String objectType = mie.getObjectType();
+    if ("Space".equals(objectType)) {
+      // check if space is allowed to current user
+      try {
+        return Administration.get().isSpaceAvailable(mie.getObjectId(), userId);
+      } catch (Exception ignored) {
+        return false;
+      }
+    } else if ("Component".equals(objectType)) {
+      // check if component is allowed to current user
+      try {
+        return Administration.get().isComponentAvailable(mie.getObjectId(), userId);
+      } catch (Exception ignored) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)